src/routes/InventoryTable.js

```diff
@@ -177,11 +178,23 @@ const Inventory = ({
   useEffect(() => {
     setActiveTabKey(initialActiveTabKey);
   }, [pathname]);
+  const searchParams = useLocation();
+  const [prm, setPrm] = useState('');
+  const [currentTab, setCurrentTab] = useState(activeTabKey);
   const handleTabClick = (_event, tabIndex) => {
-    const tabPath = tabsPath[tabIndex];
-    if (tabPath !== undefined) {
-      navigate(`${tabPath}`);
+    setCurrentTab(tabIndex);
+
+    if (currentTab !== tabIndex) {
+      setPrm(searchParams.search);
+      let tabPath = tabsPath[tabIndex];
+      if (tabPath !== undefined) {
+        if (tabPath === '') {
+          tabPath = '/';
+        }
+        history.push(`${tabPath}${prm}`);
```

Comment:
Please use [InsightsNavigate](https://github.com/RedHatInsights/insights-inventory-frontend/commit/2d9450e7427b8d37b96ddc3c77d693ed935ec4c4#diff-0142c446b1a05f59efb5cca9c6a920822b2dbecac00e598399ba2f2c3015606cR4) for the client-side routing as the router upgrade has been merged to the stage.